src/main/java/rx/exceptions/CompositeException.java

```diff
@@ -293,4 +292,25 @@ public String getMessage() {
             }
         }
     }
+
+    /**
+     * Returns the root cause of {@code e}. If {@code e.getCause()} returns {@null}, just return {@code e} itself.
+     *
+     * @param e the {@link Throwable} {@code e}.
+     * @return The root cause of {@code e}. If {@code e.getCause()} returns {@null}, just return {@code e} itself.
+     */
+    private Throwable getRootCause(Throwable e) {
+        Throwable root = e.getCause();
+        if (root == null) {
+            return e;
+        } else {
+            while(true) {
+                if (root.getCause() == null) {
```

Comment:
I was concerned it too but didn't do it because the default `getCause` will return `null` when the inner `cause` field is `this`.

However, `getCause` can be override so it's still possible. Will update it.